features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/ExploreMapFragment.kt

```diff
@@ -192,82 +177,98 @@ class ExploreMapFragment: SupportMapFragment() {
+                    .listener(object : RequestListener<Bitmap> {
+                        override fun onLoadFailed(e: GlideException?, model: Any?, target: Target<Bitmap>?,
+                                                  isFirstResource: Boolean): Boolean {
+                            Log.i("GlideException","${e?.message}")
+                            if (item.latitude != null && item.longitude != null && markerDrawable != null) {
+                                lifecycleScope.launch {
+                                    addMarkerItemToMap(item.latitude!!, item.longitude!!, markerDrawable!!, item.id)
+                                }
+                            }
+                            return false
+                        }
 
-                    override fun onResourceReady(resource: Bitmap?, model: Any?, target: Target<Bitmap>?, dataSource: DataSource?, isFirstResource: Boolean): Boolean {
-                        Log.i(this@ExploreMapFragment::class.java.simpleName, "Resource loaded")
-                        return false
-                    }
-                })
-                .submit(markerSize, markerSize)
-        }.toMutableList()
-
-        val exploreMarkers = items.zip(futureTarget).map { pair ->
-            pair.first.latitude?.let {
-                pair.first.longitude?.let { it1 ->
-                    ExploreMarkerItemUI(
-                        pair.first.id,
-                        it, it1,
-                        try {
-                            pair.second.get()
-                        } catch (e: Exception) {
-                            markersGlideRequestManager.asBitmap().load(R.drawable.ic_merchant)
-                                .submit(exploreMarkerSize, exploreMarkerSize).get()
+                        override fun onResourceReady(resource: Bitmap?, model: Any?, target: Target<Bitmap>?,
+                                                     dataSource: DataSource?, isFirstResource: Boolean): Boolean {
+                            Log.i(this@ExploreMapFragment::class.java.simpleName, "Resource loaded")
+
+                            if (item.latitude != null && item.longitude != null && resource != null) {
+                                lifecycleScope.launch {
+                                    addMarkerItemToMap(item.latitude!!, item.longitude!!, resource, item.id)
+                                }
+                            }
+                            return false
                         }
-                    )
-                }
-            }
+                    }).submit(markerSize, markerSize)
         }
+    }
 
-        futureTarget.forEach { markersGlideRequestManager.clear(it) }
-
-        withContext(Dispatchers.Main){
-            exploreMarkers.forEach{
-                it?.let { it1 -> addMarkerItemToMap(it1.latitude, it.longitude, it.logoUrl, it.id) }
+    private fun removeOldMarkers(newItems: List<SearchResult>) {
```

Comment:
Instead of resetting the map on every collection change, we calculate the difference and only render new and remove old markers.